cclunit-framework-source/src/test/ccl/ut_cclut_execute_program_with_mocks_happy.prg

```diff
@@ -78,4 +78,7 @@ end
 
 call internalSubroutine(null)
 
+set internalVariable = 1
```

Comment:
I suppose internalVariable and internalRecord->item are used to confirm that the SUT was actually executed.
Is there a situation where that is questionable? 
Does it not suffice to just use one thing for this?

Note that the adjective "internal" doesn't really fit here. Maybe "external" or "provided".

Reply:
> I suppose internalVariable and internalRecord->item are used to confirm that the SUT was actually executed.

Actually, I was doing it to comply with the statement from here:
https://github.com/cerner/cclunit-framework/pull/7/files#r250405913

> Just for kicks, I would probably demonstrate something similar for record PUBLIC::rec and variable PUBLIC::var defined by SUT.

Although reading it again, it sounds like you wanted the declarations in SUT.  Do you want me to just update these tests to move the declarations inside SUT?  And, if so, should I leave it as "internal" or do you still want "external"/"provided"?

Reply:
Correct.
That comment intended for the_script to declare a record and a variable to be overridden by the test.   I am not sure how to derive value from this capability, but I believe we do support it.  

To do this, the _namespace test would declare competing mock::rec and mock::var that would be updated rather than the public::rec and pulic::var declared by SUT.  If SUT declares the  public:: versions 'with protect', they will fall out of scope after the script returns (i.e., they will fail a validate check) yet the values will be set on the mock:: versions declared by the test.

The 'with persistscript' check needs to be performed on yet some other thing declared within SUT and only needs to be checked by one test after SUT returns I imagine.

Reply:
Sounds good.  I'll update the _namespace test and update the _happy test to include a persistscript check.